### DISCUSSION TITLE
Nice update!!

### DISCUSSION THREAD

**Discussion by kotsarvs** [theoephraim/node-google-spreadsheet#695] [General]
Nice update guys, full broke auth, not work legacy code, good joob guy's, good job!
I wrote the code last year, now it is completely broken, from authorization to deleting and adding lines, thank you for making such wonderful updates, now I need to write the code completely from scratch.

**Comment by theoephraim** [comment]
Libraries need to change over time, for security, convenience, adapt to modern standards. Open source maintainers do this work FOR FREE. I understand you were frustrated... But you're free to not use the library at all, or continue to use an old version.
